### PR TITLE
Promote mz_cluster_replica_sizes from mz_internal to mz_catalog

### PR DESCRIPTION
**Pull request by chaas**
See https://github.com/MaterializeInc/materialize/issues/26026.
We will need to coordinate this rollout with:
* Docs changes
* Console
* Cloud prometheus exporter
* mzadmin
* Demos

(see full list on tracking ticket).
Also needs a release note, since this may break users' queries (permitted because this is a catalog table in mz_internal).

### Motivation

  * This PR adds a known-desirable feature. https://github.com/MaterializeInc/materialize/issues/26026

### Tips for reviewer
* Do we need a new/different OID for the table and index, or can we reuse the existing one?
* Do we need any sort of migration for this?

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - Move the `mz_cluster_replica_sizes` system catalog table from the `mz_internal` schema to the `mz_catalog` schema, making the table definition stable. Any queries referencing the `mz_internal.mz_cluster_replica_sizes` catalog table will now need to use `mz_catalog.mz_cluster_replica_sizes`.

### CODE REVIEW

**Review by benesch — APPROVED** [review]
Exciting!

File: doc/user/content/sql/system-catalog/mz_internal.md [hunk 1]
```diff
@@ -191,7 +191,6 @@ The values in this table may change at any time. You should not rely on them for
 any kind of capacity planning.
 {{< /warning >}}
 
-<!-- RELATION_SPEC mz_internal.mz_cluster_replica_sizes -->
 | Field                  | Type        | Meaning                                                                                                                                                      |
```
Comment (ParkMyCar) [inline doc/user/content/sql/system-catalog/mz_internal.md:194]: Can you move this table description into `mz_catalog.md`? Then running `ci/test/lint-catalog-docs.sh` should result in the test from `autogenerated/mz_internal.slt` getting re-created in `autogenerated/mz_catalog.slt`

**Review by ParkMyCar — APPROVED** [review]
One comment, but otherwise if we're okay with breaking user's queries then this looks good to me!

File: doc/user/content/sql/system-catalog/mz_internal.md (hunk 1 above)
Comment (chaas) [reply doc/user/content/sql/system-catalog/mz_internal.md:194]: We don't want the docs change to reflect until the code is released tho right? If I moved it in this PR, wouldn't the docs would change at merge time?
Was thinking I'd have a separate PR to move the table description to `mz_catalog.md` that I merge the release goes out, but definitely want a sanity check there.

**Comment by chaas** [comment]
@morsapaes FYI - we'll want a release note for this one, since it changes user-facing behavior.

File: doc/user/content/sql/system-catalog/mz_internal.md (hunk 1 above)
Comment (ParkMyCar) [reply doc/user/content/sql/system-catalog/mz_internal.md:194]: Chatted offline with @chaas, I totally agree we don't want the docs to be updated before the change goes out, so the PR as-is looks good to me. Cara will follow up and update the docs once the release goes out next week.

Comment (morsapaes) [reply doc/user/content/sql/system-catalog/mz_internal.md:194]: > We don't want the docs change to reflect until the code is released tho right? If I moved it in this PR, wouldn't the docs change at merge time?

The good news is that you can merge this along with the docs if you use the `warn-if-unreleased` shortcode, as shown [here](https://github.com/MaterializeInc/materialize/tree/main/doc/user#adding-documentation-for-new-features).

**Pull request merged**